services/nuget/nuget-v2-service-family.js

```diff
@@ -27,25 +36,55 @@ const schema = Joi.object({
   }).required(),
 }).required()
 
+const xmlSchema = Joi.object({
+  feed: Joi.object({
+    entry: Joi.object({
+      'm:properties': Joi.object({
+        'd:Version': Joi.string(),
+        'd:NormalizedVersion': Joi.string(),
+        // Resharper download counts are returning -1.
+        // https://github.com/badges/shields/issues/2921
+        'd:DownloadCount': Joi.number().required(),
+        'd:Tags': Joi.string(),
+      }),
+    }),
+  }).required(),
+}).required()
+
 async function fetch(
   serviceInstance,
-  { baseUrl, packageName, includePrereleases = false }
+  { odataFormat, baseUrl, packageName, includePrereleases = false }
 ) {
-  const data = await serviceInstance._requestJson({
-    schema,
-    url: `${baseUrl}/Packages()`,
-    options: {
-      headers: { Accept: 'application/atom+json,application/json' },
-      qs: { $filter: createFilter({ packageName, includePrereleases }) },
-    },
-  })
-
-  const packageData = data.d.results[0]
+  const url = `${baseUrl}/Packages()`
+  const qs = { $filter: createFilter({ packageName, includePrereleases }) }
+
+  let packageData
+  if (odataFormat === 'xml') {
+    const data = await serviceInstance._requestXml({
+      schema: xmlSchema,
+      url,
+      options: { qs },
+    })
+    packageData = odataToObject(data.feed.entry)
+  } else if (odataFormat === 'json') {
+    const data = await serviceInstance._requestJson({
+      schema: jsonSchema,
+      url,
+      options: {
+        headers: { Accept: 'application/atom+json,application/json' },
+        qs,
+      },
+    })
+    packageData = data.d.results[0]
+  } else {
+    throw Error(`Unsupported Atom OData format: ${odataFormat}`)
+  }
 
   if (packageData) {
     return packageData
   } else if (!includePrereleases) {
     return fetch(serviceInstance, {
+      odataFormat,
       baseUrl,
       packageName,
       includePrereleases: true,
```

Comment:
This is my first time looking at this service family so trying to get a feel for it. Is this `else if` block kinda like a retry? If we didn't get any package data, and prereleases were excluded, then try again?

I'm wondering why the clause is `!includePrereleases` but the arg is set to `true` here

Reply:
Probably deserves a comment. IIRC the idea is that if there are _only_ prereleases, we show the latest of them.